infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/SecretAsContainerResourceProvisioner.java

```diff
@@ -0,0 +1,211 @@
+      throws InfrastructureException {
+
+    final String mountPath = secret.getMetadata().getAnnotations().get(ANNOTATION_MOUNT_PATH);
+    if (mountPath == null) {
+      throw new InfrastructureException(
+          format(
+              "Unable to mount secret '%s': it has to be mounted as file, but mountPath is not specified."
+                  + "Please make sure you specified '%s' annotation of secret.",
+              secret.getMetadata().getName(), ANNOTATION_MOUNT_PATH));
+    }
+
+    Volume volumeFromSecret =
+        new VolumeBuilder()
+            .withName(secret.getMetadata().getName())
+            .withSecret(
+                new SecretVolumeSourceBuilder()
+                    .withNewSecretName(secret.getMetadata().getName())
+                    .build())
+            .build();
+
+    for (PodData podData : env.getPodsData().values()) {
+      if (!podData.getRole().equals(PodRole.DEPLOYMENT)) {
+        continue;
+      }
+      if (podData
+          .getSpec()
+          .getVolumes()
+          .stream()
+          .anyMatch(v -> v.getName().equals(volumeFromSecret.getName()))) {
+        throw new InfrastructureException(
+            format(
+                "Volume name '%s' provisioned from secret, clashes with existing volume name.",
+                volumeFromSecret.getName()));
+      }
+
+      podData.getSpec().getVolumes().add(volumeFromSecret);
+
+      for (Container container : podData.getSpec().getContainers()) {
+        if (targetContainerName != null && !container.getName().equals(targetContainerName)) {
+          continue;
+        }
+        if (container
+            .getVolumeMounts()
+            .stream()
+            .anyMatch(vm -> vm.getMountPath().equals(mountPath))) {
+          throw new InfrastructureException(
+              format(
+                  "Volume Mount path '%s' provisioned from secret, clashes with existing volume mount path.",
+                  mountPath));
+        }
+        container
+            .getVolumeMounts()
+            .add(
+                new VolumeMountBuilder()
+                    .withName(volumeFromSecret.getName())
+                    .withMountPath(mountPath)
+                    .withReadOnly(true)
+                    .build());
+      }
+    }
+  }
+
+  private String envName(Secret secret, String key) throws InfrastructureException {
+    String mountEnvName;
+    if (secret.getData().size() == 1) {
+      try {
+        mountEnvName =
+            firstNonNull(
+                secret.getMetadata().getAnnotations().get(ANNOTATION_ENV_NAME),
+                secret
+                    .getMetadata()
+                    .getAnnotations()
+                    .get(format(ANNOTATION_ENV_NAME_TEMPLATE, key)));
+      } catch (NullPointerException e) {
+        throw new InfrastructureException(
+            format(
+                "Unable to mount secret '%s': it has to be mounted as Env, but env name is not specified."
```

Comment:
Unable to mount the secret '%s'. It is configured to be mounted as an environment variable but the name of the environment variable is not specified. Please define the '%s' annotation on the secret to specify it.